drivers/VDI.py

```diff
@@ -576,6 +576,20 @@ def in_sync_with_xenapi_record(self, x):
             return False
         return True
 
+    def _list_vdi_snapshots(self, vdi_uuid):
+        """List UUIDs of all direct snapshots of a VDI"""
+        try:
+            vdi_ref = self.session.xenapi.VDI.get_by_uuid(vdi_uuid)
+            record = self.session.xenapi.VDI.get_record(vdi_ref)
+            return [
+                self.session.xenapi.VDI.get_uuid(ref)
+                for ref in record.get("snapshots", [])
+                if self.session.xenapi.VDI.get_uuid(ref)
```

Comment:
Two call for the XAPI is not optimal but also, this will raise if ref is not a correct OpaqueRef